src/CodeGenerator/AbstractInjector.php

```diff
@@ -49,14 +49,21 @@ public function __construct(InjectorInterface $injector, ContainerInterface $con
      */
     abstract protected function loadFactoryList() : void;
 
+    private function instantiateFactory(string $factoryClass) : FactoryInterface
+    {
+        return new $factoryClass();
+    }
+
     private function getFactory($type) : FactoryInterface
     {
-        if (is_string($this->factories[$type])) {
-            $factory = $this->factories[$type];
-            $this->factories[$type] = new $factory();
+        $factory = $this->factories[$type];
+
+        if (is_string($factory)) {
+            $factory = $this->instantiateFactory($factory);
+            $this->factories[$type] = $factory;
```

Comment:
Can be simplified to `return $this->factories[$type] = $this->instantiateFactory($factory)`.

Also, `$factory` changing type in here is very messy...

Reply:
I know. Need to go to the think tank on how to change this without breaking BC...

Reply:
Two arrays, one of which is for instantiated factories